File: VB/19_OLE/GetEmbeddedFileName.vb
Imports Spire.Presentation
Imports System.ComponentModel
Imports System.Data.SqlTypes
Imports System.IO
Imports System.Text


Namespace GetEmbeddedFileName
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load document from disk
			presentation.LoadFromFile("..\..\..\..\..\..\Data\oleTest.pptx")

			'Loop through the slides and shapes
			For Each slide As ISlide In presentation.Slides
				For Each shape As IShape In slide.Shapes
					If TypeOf shape Is IOleObject Then
						'Find OLE object
						Dim oleObject As IOleObject = TryCast(shape, IOleObject)
						'Get OLE object label name
						Dim oleFileName As String = oleObject.EmbeddedFileName
						MessageBox.Show("The name of the OLE object label is:" & oleFileName)
					End If
				Next shape
			Next slide



		End Sub
	End Class
End Namespace

File: CS/19_OLE/GetEmbeddedFileName.cs
using Spire.Presentation;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.SqlTypes;
using System.Drawing;
using System.IO;
using System.Text;
using System.Windows.Forms;


namespace GetEmbeddedFileName
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {         
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load document from disk
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\oleTest.pptx");

            //Loop through the slides and shapes
            foreach (ISlide slide in presentation.Slides)
            {
                foreach (IShape shape in slide.Shapes)
                {
                    if (shape is IOleObject)
                    {
                        //Find OLE object
                        IOleObject oleObject = shape as IOleObject;
                        //Get OLE object label name
                        string oleFileName = oleObject.EmbeddedFileName;
                        MessageBox.Show("The name of the OLE object label is:"+oleFileName);
                    }
                }
            }
         
            
          
        }
    }
}
